src/Components/I18nWidget.tsx

```diff
@@ -27,7 +27,7 @@ export const I18nWidget = (props: I18nWidgetProps) => {
         ),
         negativeText: <Text id="stay" />,
         onNegativeFeedback: () => {
-            setNewLanguage(savedLocale);
+            setNewLanguage(savedLocale); // TODO: Is this call needed?
```

Comment:
Yes, it is, but I was confused, too, so maybe let's make this a little more verbose:
```suggestion
			// Reset the select to show the current language instead of the just selected one.
            setNewLanguage(savedLocale); 
```